frontend/src/views/components/Legend/LegendTimeline/LegendTimeline.component.tsx

```diff
@@ -0,0 +1,51 @@
+import Datepicker from 'views/shared/datepicker';
+import { useCallback } from 'react';
+import type { Timeline } from 'types/layer';
+import { T } from '@transifex/react';
+
+interface LegendTimelineProps {
+  timeline: Timeline;
+  layerName: string;
+  setDate: (date: Date) => void;
+  selected: Date;
+}
+
+const LegendTimeline = ({ timeline, layerName, setDate, selected }: LegendTimelineProps) => {
+  const { startDate, endDate, period, steps } = timeline;
+
+  const getResolution = useCallback(() => {
+    const resolutionPeriods = {
+      yearly: 'year',
+      monthly: 'month',
+      daily: 'day',
+    };
+    return resolutionPeriods[period];
+  }, [period]);
+
+  return (
+    <>
+      <div className="m-legend-timeline">
+        <label htmlFor="timeline-datepicker" className="timeline-title">
```

Comment:
You might have several layers with a datepicker. You need a unique value for `htmlFor` and the datepicker's `id`.